site/en/docs/extensions/reference/offscreen/index.md

```diff
@@ -40,44 +40,54 @@ Use `chrome.offscreen.createDocument()` and `chrome.offscreen.closeDocument()` f
 
 ```js
 chrome.offscreen.createDocument({
-  url: chrome.runtime.getURL('off_screen.html'),
+  url: 'off_screen.html',
   reasons: ['CLIPBOARD'],
   justification: 'reason for needing the document',
 });
 
 chrome.offscreen.closeDocument()
 ```
 
-The following example shows how to check for existing offscreen documents. The `hasOffscreenDocument()` function uses [clients.matchAll()](https://developer.mozilla.org/docs/Web/API/Clients/matchAll) to find existing offscreen documents. Please note: the function assumes that your extension uses a single offscreen document. 
+The following example shows how to ensure that offscreen documents have already been created. The `setupOffscreenDocument()` function uses [clients.matchAll()](https://developer.mozilla.org/docs/Web/API/Clients/matchAll) to find existing offscreen documents. If no existing offscreen documents, create it. Please note: the function assumes that your extension uses a single offscreen document. 
 
 ```js
-async function hasOffscreenDocument(path) {
+let creating; // A global promise to avoid concurrency issues
+async function setupOffscreenDocument(path) {
   // Check all windows controlled by the service worker to see if one 
   // of them is the offscreen document with the given path
+  let exist = false;
   const offscreenUrl = chrome.runtime.getURL(path);
   const matchedClients = await clients.matchAll();
   for (const client of matchedClients) {
     if (client.url === offscreenUrl) {
-      return true;
+      exist = true;
+      break;
+    }
+  }
+  
+  // create offscreen document if it's not existing
+  if (!exist) {
+    if (creating) {
+      await creating;
```

Comment:
Do we need to `await` here? We could also simply return `creating` and avoid the `else` block below.

Reply:
I prefer the current way here, which looks more symmetrical